crates/uv/src/commands/project/remove.rs

```diff
@@ -93,29 +95,30 @@ pub(crate) async fn remove(
     }?;
 
     for package in packages {
+        let requirement = Requirement::from_str(package.as_str())?;
```

Comment:
I don't know if we can avoid doing this conversion here. All of the `remove_dependency` functions rely on the `find_dependencies` function which requires the `Requirement` type. `\cc @zanieb

Reply:
From a quick scan of the changes, it seems like there are a couple alternative options:

1. Perform marker filtering on the result of `find_dependencies` when relevant or...
2. Update `find_dependencies` to take `Option<Marker>` in addition to `PackageName`

Reply:
I've gone with the second option